Menus/BuyMenuGUI.cpp

```diff
@@ -1208,20 +1208,45 @@ void BuyMenuGUI::Update()
 
         // Get handle to the currently selected item, if any
         GUIListPanel::Item *pItem = m_pShopList->GetItem(m_ListItemIndex);
+        std::string description = "";
 
-        // Show popup info box next to selected item, but only if it has a description
-        string description = "";
-        // Get it from the regular Entitiy preset
-        if (pItem && pItem->m_pEntity && !pItem->m_pEntity->GetDescription().empty())
+        if (pItem && pItem->m_pEntity) {
             description = pItem->m_pEntity->GetDescription();
-        // Show popup info box next to selected module item, but only if it has a description
-        else if (pItem && pItem->m_ExtraIndex >= 0)
-        {
-            const DataModule *pModule = g_PresetMan.GetDataModule(pItem->m_ExtraIndex);
-            if (pModule && !pModule->GetDescription().empty())
+            const Entity* currentItem = pItem->m_pEntity;
+            const ACraft* pItemAsCraft = dynamic_cast<const ACraft*>(currentItem);
```

Comment:
Naming, we only include prefixes for member variables, and never the `p` pointer prefix. Intellisense and context can tell you if a variable is a pointer just fine.

Applies throughout your changes.

Reply:
Should this include the preexisting `pItem` variable? I've left it for the moment as it'll take a full refactor of `BuyMenu.cpp` to remove.

Reply:
Yes, this should include pItem, but as you say, that can wait til a full refactor happens.